packages/langium/src/lsp/language-server.ts

```diff
@@ -82,23 +81,33 @@ export function startLanguageServer(services: LangiumServices): void {
     connection.listen();
 }
 
-export function addDocumentsHandler(documents: TextDocuments<TextDocument>, services: LangiumServices): void {
+export function addDocumentsHandler(connection: Connection, documents: TextDocuments<TextDocument>, services: LangiumServices): void {
     const documentBuilder = services.documents.DocumentBuilder;
     let changeTokenSource: AbstractCancellationTokenSource;
     let changePromise: Promise<void> | undefined;
-    documents.onDidChangeContent(async change => {
+
+    async function onDidChange(changed: URI[], deleted?: URI[]): Promise<void> {
         changeTokenSource?.cancel();
         if (changePromise) {
             await changePromise;
         }
         changeTokenSource = startCancelableOperation();
-        try {
-            changePromise = documentBuilder.documentChanged(URI.parse(change.document.uri), changeTokenSource.token);
-        } catch (err) {
-            if (err !== OperationCancelled) {
-                throw err;
-            }
-        }
+        changePromise = documentBuilder
+            .update(changed, deleted ?? [], changeTokenSource.token)
+            .catch(err => {
```

Comment:
Why did you change to a callback instead of a `catch` block?

Reply:
Because the `catch` block doesn't actually catch the exception there (since we don't await the promise). It only catches it if we are at the previous `await` instruction, but there's no guarantee that we actually hit that code.